google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherClientSnippets.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.examples.pubsub.snippets;
+
+import com.google.cloud.Identity;
+import com.google.cloud.Role;
+import com.google.cloud.pubsub.spi.v1.PagedResponseWrappers.ListTopicSubscriptionsPagedResponse;
+import com.google.cloud.pubsub.spi.v1.PagedResponseWrappers.ListTopicsPagedResponse;
+import com.google.cloud.pubsub.spi.v1.PublisherClient;
+import com.google.iam.v1.Binding;
+import com.google.iam.v1.Policy;
+import com.google.iam.v1.TestIamPermissionsResponse;
+import com.google.pubsub.v1.ListTopicSubscriptionsRequest;
+import com.google.pubsub.v1.ListTopicsRequest;
+import com.google.pubsub.v1.ProjectName;
+import com.google.pubsub.v1.Topic;
+import com.google.pubsub.v1.TopicName;
+import java.util.LinkedList;
+import java.util.List;
+
+/** This class contains a number of snippets for the {@link PublisherClient} interface. */
+public class PublisherClientSnippets {
+
+  private final String projectId;
+
+  public PublisherClientSnippets(String projectId) {
+    this.projectId = projectId;
+  }
+
+  /** Example of creating a topic. */
+  public Topic createTopic(String topicName) throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      // [START createTopic]
+      TopicName formattedTopicName = TopicName.create(projectId, topicName);
+      Topic topic = publisherClient.createTopic(formattedTopicName);
+      // [END createTopic]
+      return topic;
+    }
+  }
+
+  /** Example of listing topics, specifying the page size. */
+  public ListTopicsPagedResponse listTopics() throws Exception {
+    try (PublisherClient publisherClient = PublisherClient.create()) {
+      // [START listTopics]
+      ListTopicsRequest listTopicsRequest =
+          ListTopicsRequest.newBuilder()
+              .setProjectWithProjectName(ProjectName.create(projectId))
+              .setPageSize(100)
```

Comment:
This seems to be the basic example for `listTopics` - why not make it minimal and omit the page size? It looks like all of the examples set page size, in fact; I don't think think is a common use case, especially if the flattened iteration is used, where the user doesn't have to think about the individual pages.